fmt_xlong.c

```diff
@@ -0,0 +1,17 @@
+#include "fmt.h"
+
+/* writes ulong u in hex to char *s, does not NULL-terminate */
+unsigned int fmt_xlong(char *s, unsigned long u)
+{
+ unsigned int len;
+ unsigned long q;
+ unsigned long c;
+ len = 1; q = u;
+ while (q > 15) { ++len; q /= 16; }
+ if (s)
```

Comment:
Please reformat this block: put the { in the line of the if() and unwrap the do-while loop.